src/librustc_codegen_ssa/back/link.rs

```diff
@@ -1179,6 +1179,34 @@ fn add_pre_link_args(
     cmd.args(&sess.opts.debugging_opts.pre_link_args);
 }
 
+/// Add an LLD link script embedded in the target, if applicable.
+fn add_lld_link_script(
+    cmd: &mut dyn Linker,
+    sess: &Session,
+    flavor: LinkerFlavor,
+    tmpdir: &Path,
+    crate_type: CrateType,
+) {
+    match (flavor, crate_type, &sess.target.target.options.lld_link_script) {
+        (
+            LinkerFlavor::Lld(LldFlavor::Ld),
+            CrateType::Cdylib | CrateType::Executable,
+            Some(script),
+        ) => {
+            let file_name = ["rustc", &sess.target.target.llvm_target, "linkfile.ld"].join("-");
+
+            let path = tmpdir.join(file_name);
+            if let Err(e) = fs::write(&path, script) {
+                sess.fatal(&format!("failed to write link script to {}: {}", path.display(), e));
+            }
+
+            cmd.arg("--script");
+            cmd.arg(path);
+        }
+        _ => {}
+    }
+}
```

Comment:
This could be generalized to work with `LinkerFlavor::Ld` as well (which should require no changes from what you've got already).

Also it would be good to emit an error for invalid configurations (for example when the linker is actually `link.exe`) instead of doing nothing.

Reply:
What about platforms such as macOS, where `ld` does not support link scripts?

Reply:
I *think* you can detect that via the `linker_is_gnu` flag

Reply:
Got it, looks much cleaner this way